### PR TITLE
Onboard Argus Scrape Configs: update command

### PR DESCRIPTION
**Pull request by DiogoFerrao**

### CODE REVIEW

File: internal/cmd/argus/scrape-config/update/update.go [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+}
+
+func NewCmd(p *print.Printer) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   fmt.Sprintf("update %s", jobNameArg),
+		Short: "Updates a scrape configuration of an Argus instance",
+		Long: fmt.Sprintf("%s\n%s\n%s",
+			"Updates a scrape configuration of an Argus instance.",
+			"The payload can be provided as a JSON string or a file path prefixed with \"@\".",
+			"See https://docs.api.stackit.cloud/documentation/argus/version/v1#tag/scrape-config/operation/v1_projects_instances_scrapeconfigs_partial_update for information regarding the payload structure.",
```
Comment (joaopalet) [inline internal/cmd/argus/scrape-config/update/update.go:41]: I think you meant
```suggestion
			"See https://docs.api.stackit.cloud/documentation/argus/version/v1#tag/scrape-config/operation/v1_projects_instances_scrapeconfigs_update for information regarding the payload structure.",
```
right?

File: internal/cmd/argus/scrape-config/update/update.go [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+		),
+		Args: args.SingleArg(jobNameArg, nil),
+		Example: examples.Build(
+			examples.NewExample(
+				`Update a scrape configuration from Argus instance "xxx", using an API payload sourced from the file "./payload.json"`,
```
Comment (joaopalet) [inline internal/cmd/argus/scrape-config/update/update.go:46]: Do you think it's relevant to add `Update a scrape configuration with job name "my-config" ...`?

File: internal/cmd/argus/scrape-config/update/update.go [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+			}
+
+			// Call API
+			req := buildRequest(ctx, model, apiClient)
+			_, err = req.Execute()
+			if err != nil {
+				return fmt.Errorf("update scrape config: %w", err)
+			}
+
+			// The API has no status to wait on, so async mode is default
```
Comment (joaopalet) [inline internal/cmd/argus/scrape-config/update/update.go:85]: Does the Portal also indicate the update as done immediatelly?

File: internal/cmd/argus/scrape-config/update/update_test.go [hunk 4]
```diff
@@ -0,0 +1,294 @@
...
+		{
+			description: "instance id invalid 2",
+			argValues:   fixtureArgValues(),
+			flagValues: fixtureFlagValues(func(flagValues map[string]string) {
+				flagValues[instanceIdFlag] = "invalid-uuid"
+			}),
+			isValid: false,
+		},
+		{
+			description: "invalid json",
```
Comment (joaopalet) [inline internal/cmd/argus/scrape-config/update/update_test.go:205]: Missing a test case for just the `payload` flag missing

File: internal/cmd/argus/scrape-config/update/update.go (hunk 3 above)
Comment (DiogoFerrao) [reply internal/cmd/argus/scrape-config/update/update.go:85]: This is something that was also commented on the Terraform provider, but I looked into it now and the endpoint seems to be synchronous. Removed the comment

Comment (joaopalet) [reply internal/cmd/argus/scrape-config/update/update.go]: If we know the endpoint is synchronous then we can remove the comment, if it's just fast I think we could keep it

Comment (DiogoFerrao) [reply internal/cmd/argus/scrape-config/update/update.go]: The response code is a 202, which would indicate asynchronous. So maybe it's just fast, although the message in the reply states "Successfully updated scrape config".

Comment (DiogoFerrao) [reply internal/cmd/argus/scrape-config/update/update.go:85]: I'll add the comment back

**Pull request merged**